File: app/src/main/java/com/example/moneymanager/GamingRoomAdapter.java
package com.example.moneymanager;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class GamingRoomAdapter extends RecyclerView.Adapter<GamingRoomAdapter.ViewHolder> {
    private ArrayList<ActivityList> mActivityList;

    public GamingRoomAdapter(ArrayList<ActivityList> activityList) {
        this.mActivityList = activityList;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public ImageView activityImages;
        public TextView activityNames, activityDates, balance;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            activityImages = itemView.findViewById(R.id.ivActivityLogos);
            activityNames = itemView.findViewById(R.id.tvActivityNames);
            activityDates = itemView.findViewById(R.id.tvActivityDates);
            balance = itemView.findViewById(R.id.tvActivity_balance);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        // Inflate the custom layout
        View gamingView = inflater.inflate(R.layout.rv_gaming_items,
                parent, false);

        // Return a new holder instance
        GamingRoomAdapter.ViewHolder viewHolder = new ViewHolder(gamingView);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ActivityList activityList = mActivityList.get(position);

        holder.activityImages.setImageResource(activityList.getActivityIcons());
        if (position == 0) {
            holder.activityImages.setPadding(5,5,5,5);
        }
        holder.activityNames.setText(activityList.getActivityNames());
        holder.activityDates.setText(activityList.getActivityDates());
        holder.balance.setText(activityList.getBalance());

    }

    @Override
    public int getItemCount() {
        return mActivityList.size();
    }
}

File: app/src/main/java/com/example/moneymanager/GamingRoomAdapter.kt
package com.example.moneymanager

import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.example.moneymanager.databinding.RvGamingItemsBinding
import com.example.moneymanager.model.Activity

class GamingRoomAdapter(private val activities: List<Activity>) : RecyclerView.Adapter<GamingRoomAdapter.ViewHolder>() {
    inner class ViewHolder(val gamingItemsBinding: RvGamingItemsBinding) : RecyclerView.ViewHolder(gamingItemsBinding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        // Inflate the custom layout
        val binding = RvGamingItemsBinding.inflate(
            LayoutInflater.from(parent.context),
                parent, false)

        // Return a new holder instance
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        with(holder) {
            with(activities[position]) {
                gamingItemsBinding.ivActivityLogos.setImageResource(icon)
                if (position == 0) {
                    gamingItemsBinding.ivActivityLogos.setPadding(5, 5, 5, 5)
                }
                gamingItemsBinding.tvActivityNames.text = this.name

                gamingItemsBinding.tvActivityDates.text = this.date

                gamingItemsBinding.tvActivityBalance.text = this.balance
            }
        }
    }

    override fun getItemCount(): Int {
        return activities.size
    }
}

